synfig-studio/src/synfigapp/canvasinterface.cpp

```diff
@@ -699,6 +699,45 @@ CanvasInterface::jump_to_prev_keyframe()
 	//catch(...) { synfig::warning("Unable to find prev keyframe"); }
 }
 
+static void update_layer_size(CanvasInterface& canvas_interface, Layer::Handle& layer, bool resize_image) {
```

Comment:
It looks like better to pass `RendDesc` isntead of `CanvasInterface` here.

Reply:
Agree
Because that function is internal and used only within synfigapp.

[C++ best practices](https://lefticus.gitbooks.io/cpp-best-practices/content/08-Considering_Performance.html) **Reduce Temporary Objects** section may also be helpful to validate why this is better :)

Reply:
Done